src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -43,7 +42,9 @@ internal sealed partial class Http2Connection : HttpConnectionBase, IDisposable
         private int _maxConcurrentStreams;
         private int _pendingWindowUpdate;
         private long _idleSinceTickCount;
-        private int _pendingWriters;
+
+        private readonly Queue<WriteQueueEntry> _writeQueue = new Queue<WriteQueueEntry>();
+        private TaskCompletionSource? _writeSignal;
```

Comment:
Same with this one; it'd be good to experiment with making this reusable.  This one should be pretty easy.

Reply:
Yeah, this one is straightforward.

Reply:
But, it goes away if we use `Channel<T>`